### ISSUE TITLE
"Prefers contrast" admin theming

### ISSUE THREAD

**Issue by thibaudcolas** [wagtail/wagtail#12176]
### Is your proposal related to a problem?

From feedback by users with low vision:

> One suggestion made if there was a battle between general UI design and accessibility would be to have a switch to enable a more accessible version vs a more simplified version.

This is in the context of their other feedback about UI elements lacking affordances, some of which I’ve asked for follow-up info about:

- #9498 
- _Collapsibility options were not immediately apparent_
- _Text box edges weren’t clearly demarcated, some users had no idea where they were._
- #12175 
- #12126

### Describe the solution you'd like

We should consider a "higher visual affordances" theming option in the CMS, to go alongside recent work on dark mode (#10056) and information density (#11265). 

Like our other theming options, this could work well with the [prefers-contrast](https://drafts.csswg.org/mediaqueries-5/#prefers-contrast) media query. Official description, emphasis mine:

> The prefers-contrast media feature is used to detect if the user has requested more or less contrast in the page. This could be responded to, for example, by adjusting the contrast ratio between adjacent colors, **or by changing how much elements stand out visually, such as by adjusting their borders**.

We could use this to do color theming (more or less contrast), and also to offer an opt-in way to:

1. Add extra borders around interactive UI elements (similarly to `forced-colors` mode)
2. Switch our last few "hover-only" elements to "always-visible" (comment icons, anchor links, etc)

### Describe alternatives you've considered

I’ve considered other variants on an "accessibility mode", but would like to steer clear of any risk we would compromise on the usability or accessibility of the UI in its "default" variant. 

In my mind, this mode would be a way for users to opt into higher-contrast versions of UI elements that would be an improvement for some users but not all (again similarly to forced colors mode).

### Working on this

This requires validation and research by the @wagtail/accessibility team and other accessibility and UI contributors. In particular:

- [x] Confirmation that the concept seems sensible
- [x] Review of best practices to implement `prefers-contrast` theming across existing apps

#### Design & implementation tasks

- [ ] Review of which aspects of the CMS interface could be candidates to adjust like this (which elements, which properties of elements) @albinazs @Toriasdesign 
- [ ] UX prototypes / design recommendations on how to adjust relevant UI elements
- [ ] UX of theming control in user profile: confirm field label, options, help text
- [ ] Implement new user profile field
- [ ] Replicate the extra borders added in "forced colors" theming, for this new "prefers contrast" theming

**Comment by thibaudcolas** [comment]
I’ve updated the ticket based on recent accessibility team discussions. In particular:

- We think this will be a worthwhile addition.
- We expect it will be clearer as a separate theming option, rather than integrating it with the current "Admin theme" control.
  - This will allow us to better explain what the field does (good label, possibly help text)
  - This will make for a much simpler mapping to the `prefers-contrast` media query
- We want to consider having both "more" contrast and "less" contrast options (see [Sara Joy’s thoughts on contrast levels](https://sarajoy.dev/blog/a11y-aspects/#contrast-levels)).

**Comment by thibaudcolas** [comment]
Adding notes about similar settings elsewhere.

- Windows has its well-known [contrast themes](https://support.microsoft.com/en-gb/windows/change-color-contrast-in-windows-fedc744c-90ac-69df-aed5-c8a90125e696), with "high contrast" and "low contrast" built-in options, and support for users to create custom themes.
- macOS has "Increase contrast" on-off, which increases the color contrast of elements and introduces extra borders (see screen recording below)
- iOS has separate on-off settings, "Button shapes" (adds extra underlines and borders), "On-off labels" (add labels for switches), "Increase contrast" (color contrast only)

**Comment by thibaudcolas** [comment]
Based on the above research, I’ve updated the description to start defining more specific design and implementation tasks.

For the new user profile field, I would recommend the following:

- Type: `<select>`
- Label: "Contrast preferences"
- Options: "System default" (default, would respect the `prefers-contrast` media query); "More contrast". This would make it possible to introduce a "Less contrast" option in the future.

This is very similar to the "Admin theme" and "Density" controls implemented a few versions ago (#11864).

For the actual adjustments to the UI, I think a solid baseline would be to replicate the "forced colors mode" adjustments we already have, adding extra borders to some elements. Then we can introduce more similar changes, and decide whether they should be for forced colors mode or "prefers contrast" theming only, or both, or just all users.

**Comment by albinazs** [comment]
In addition to the changes mentioned, I plan to evaluate the color contrast itself, focusing on identifying any obvious [APCA](https://www.w3.org/WAI/GL/WCAG3/2021/how-tos/visual-contrast-of-text/) (a WIP contrast check for the upcoming WCAG 3.0) issues, or WCAG 2.2 AAA-level contrast failures in our admin interface, and consider making the necessary adjustments to ensure compliance with both standards.
